// TODO getters/setters
  function testcase() 
  {
    var accessed = false;
    function callbackfn(val, idx, obj) 
    {
      if (idx > 1)
      {
        accessed = true;
      }
    }
    var arr = [];
    arr[5] = 10;
    arr[10] = 100;
    Object.defineProperty(arr, "1", {
      get : (function () 
      {
        throw new RangeError("unhandle exception happened in getter");
      }),
      configurable : true
    });
    Object.defineProperty(arr, "2", {
      get : (function () 
      {
        accessed = true;
        return 100;
      }),
      configurable : true
    });
    try
{      arr.map(callbackfn);
      return false;}
    catch (ex)
{      return (ex instanceof RangeError) && ! accessed;}

  }
  {
    var __result1 = testcase();
    var __expect1 = true;
  }
  
